cobol-parser/src/main/scala/za/co/absa/cobrix/cobol/reader/extractors/record/RecordExtractors.scala

```diff
@@ -43,21 +43,24 @@ object RowExtractors {
     *                               All other segment redefines will be skipped.
     * @param generateInputFileField if true, a field containing input file name will be generated
     * @param inputFileName          An input file name to put if its generation is needed
-    * @return A Spark [[org.apache.spark.sql.Row]] object corresponding to the record schema.
+    * @return An Array[Any] object corresponding to the record schema.
     */
   @throws(classOf[IllegalStateException])
-  def extractRecord(ast: CopybookAST,
-                    data: Array[Byte],
-                    offsetBytes: Int = 0,
-                    policy: SchemaRetentionPolicy = SchemaRetentionPolicy.KeepOriginal,
-                    variableLengthOccurs: Boolean = false,
-                    generateRecordId: Boolean = false,
-                    segmentLevelIds: Seq[Any] = Nil,
-                    fileId: Int = 0,
-                    recordId: Long = 0,
-                    activeSegmentRedefine: String = "",
-                    generateInputFileField: Boolean = false,
-                    inputFileName: String = ""): Row = {
+  def extractRecord[T: ClassTag](
+    ast: CopybookAST,
+    data: Array[Byte],
+    offsetBytes: Int = 0,
+    policy: SchemaRetentionPolicy = SchemaRetentionPolicy.KeepOriginal,
+    variableLengthOccurs: Boolean = false,
+    generateRecordId: Boolean = false,
+    segmentLevelIds: Seq[Any] = Nil,
+    fileId: Int = 0,
+    recordId: Long = 0,
+    activeSegmentRedefine: String = "",
+    generateInputFileField: Boolean = false,
+    inputFileName: String = "",
+    handler: RecordHandler[T]
+  ): Seq[Any] = {
```

Comment:
I'm wondering what were considerations of returning `Seq[Any]` instead of `T`? My guess is maybe it is more natural for JSON and XML while still okay for Spark's `Row`.

Reply:
The issue here is that all the operations in this method create sequences such as 
```fileId +: recordId +: inputFileName +: (segmentLevelIds ++ records)```
I don't see a way to do it unless we force the user to provide a way for the `T` class to convert from the `Seq[Any]` to `Seq[T]`.  In the Spark case, this is fine since `Row` does exactly that.

Reply:
Yeah, this makes perfect sense.